client/sagas/placeSagas.js

```diff
@@ -16,8 +21,20 @@ function* fetchPlace(action) {
   }
 }
 
+function* getPlacesWithCategories() {
+  try {
+    const condition = yield select(selectCondition);
+    const places = yield call(getPlaceIds, condition);
+    const randomPlace = getRandom(places);
+    yield put(placeActions.setDetails(randomPlace));
+  } catch (e) {
+    console.log('error! ', e);
+  }
+}
+
 function* placeSagas() {
   yield takeEvery(FETCH_PLACES, fetchPlace);
+  yield takeLatest(SET_CATEGORY, getPlacesWithCategories);
```

Comment:
i don't think it's the best idea to call the api everytime the user clicks on a checkbox.